code/src/main/java/com/onlinecontacttracing/servlets/authentication/State.java

```diff
@@ -1,17 +1,39 @@
 package com.onlinecontacttracing.authentication;
-
+import com.onlinecontacttracing.authentication.AuthenticationScope;
+import java.util.List;
+import java.util.HashMap;
+import java.util.ArrayList;
 /*
  * Helper class to pass along state information in URL and servlets
  */
 class State {
   final String idToken;
   String userId;
-  final boolean calendar;
-  final boolean contacts;
+  List<AuthenticationScope> authenticationScopes;
 
-  public State(String idToken, boolean calendar, boolean contacts) {
+  public State(String idToken, List<AuthenticationScope> authenticationScopes) {
```

Comment:
Can we rename this class AuthorizationRoundTripState?

Reply:
yep